client/packages/system/src/Patient/DetailView/ProgramEnrolment/DetailModal.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { FC } from 'react';
+import {
+  DetailViewSkeleton,
+  DialogButton,
+  SaveDocumentMutation,
+  useDialog,
+  useJsonForms,
+} from '@openmsupply-client/common';
+import { useProgramEnrolment } from './api/hooks';
+import { usePatientModalStore } from '../../hooks';
+import { PatientModal } from '..';
+import { usePatient } from '../../api';
+
+const useUpsertProgramEnrolment = (
+  patientId: string,
+  type: string
+): SaveDocumentMutation => {
+  const { mutateAsync: insertProgram } = useProgramEnrolment.document.insert();
+  const { mutateAsync: updateProgramEnrolment } =
+    useProgramEnrolment.document.update();
+  return async (jsonData: unknown, formSchemaId: string, parent?: string) => {
+    if (parent === undefined) {
+      const result = await insertProgram({
+        data: jsonData,
+        schemaId: formSchemaId,
+        patientId,
+        type,
+      });
+      return result;
+    } else {
+      const result = await updateProgramEnrolment({
+        data: jsonData,
+        parent,
+        schemaId: formSchemaId,
+        patientId,
+        type,
+      });
+      return result;
+    }
+  };
+};
+
+export const ProgramDetailModal: FC = () => {
+  const patientId = usePatient.utils.id();
+
+  const { current, documentName, documentType, reset } = usePatientModalStore();
+  const handleSave = useUpsertProgramEnrolment(patientId, documentType || '');
+  const { JsonForm, isLoading } = useJsonForms(documentName, {
+    handleSave,
+    showButtonPanel: false,
+  });
+
+  const { Modal } = useDialog({
+    isOpen: current === PatientModal.Program,
+    onClose: reset,
+  });
+
+  if (isLoading) return <DetailViewSkeleton />;
+
+  return (
+    <Modal
+      title=""
```

Comment:
another TODO would be to make the title dynamic. would have to remove the current title from the ui schema and fetch the patient details to get the patient name